frontend/lib/with-admin.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react"
+import { NextPageContext as NextContext } from "next"
+import { isAdmin } from "/lib/authentication"
+import AdminError from "/components/Dashboard/AdminError"
+
+export default function withAdmin(Component: any) {
```

Comment:
At the moment `withAdmin` is always used with `withSignedIn`. While it certainly works it seems fragile to me since one could forgot to wrap the component the `withSignedIn` HOC as  well. Should `withAdmin` redirect non logged in users to the login page too?

Reply:
Yup, changed that. As these become pretty similar, I was wondering if we could do with just one common one and pass predicates to it.